app/views/grant_submissions/submissions/index.html.haml

```diff
@@ -43,22 +47,59 @@
           %th
             = sort_link(@q, :state, 'State')
           %th
-            = sort_link(@q, :updated_at, 'Last Updated')
-          %th
+            = sort_link(@q, :updated_at, 'Updated')
+          - if current_user.get_role_by_grant(grant: @grant)
+            %th
+              Reviews
+            %th
+              %span{ title: 'Average of scored reviews' }
+                Overall Impact
+            %th
+              %span{ title: 'Average of scored Criteria' }
+                Composite
+
       %tbody
         - @submissions.each do |submission|
-          %tr
+          - submission_reviews = submission.reviews.to_a
+          - has_completed_reviews = submission_reviews.any?{ |review| review.overall_impact_score.present? }
+
+          %tr{data: { submission: submission.id } }
             %td
               = link_to sortable_full_name(submission.applicant), grant_submission_path(@grant, submission)
             %td
               = link_to submission.title, grant_submission_path(@grant, submission)
             %td
-              = submission.state.titleize
-            %td
-              = submission.updated_at
+              %div.status{data: {status: submission.id} }
+                = submission.state.titleize
+              - if current_user.get_role_by_grant(grant: @grant) != 'viewer' && current_user != submission.applicant
+                %div.administer.small{data: { status: submission.id} }
+                  - if submission.draft?
+                    = link_to 'Edit', edit_grant_submission_path(@grant, submission)
+                  - elsif submission.submitted?
+                    = link_to 'Switch to Draft', unsubmit_grant_submission_path(@grant, submission), method: :patch, title: 'Allow for editing'
+                - if current_user.get_role_by_grant(grant: @grant) == 'admin' && !@grant.published?
+                  %div.delete
+                    = link_to 'Delete', grant_submission_path(@grant, submission), method: :delete, data: { confirm: 'Are you sure?'}, class: 'small', confirm: 'Are you sure?'
             %td
-              - if current_user.get_role_by_grant(grant: @grant)
-                #ul.menu.simple
-                  = render partial: "#{current_user.get_role_by_grant(grant: @grant)}_links", locals: { submission: submission, grant: @grant }
+              = date_time_separate_lines(submission.updated_at)
+            - if current_user.get_role_by_grant(grant: @grant)
+              %td
+                = link_to 'Reviews', grant_submission_reviews_path(@grant, submission)
+                - if submission.reviews.any?
+                  %div.count.small{title: 'Completed / Assigned'}
+                    #{submission_reviews.count{ |review| !review.overall_impact_score.nil? }} / #{submission_reviews.count}
```

Comment:
these calculations can probably be encapsulated in a decorator helper method which does the same calculations but not mixed with the html elements so that you can unit test it. It returns these calculated result and here you just display the result.

Reply:
here you basically show three counts: Completed / Assigned, Average of Completed, Average of Scored Criteria. So you can call a function like `stats = get_review_stats(sumission, submission_reviews)` and `stats` is a hash: `{ completed_assigned: 11, avg_completed: 22, avg_scored_criteria: 33}